### PR TITLE
feat(rdb): instance resource

### PR DESCRIPTION
**Pull request by kindermoumoute**
depends on #332

### CODE REVIEW

File: scaleway/resource_rdb_instance_beta.go [hunk 1]
```diff
@@ -0,0 +1,206 @@
...
+			"name": {
+				Type:        schema.TypeString,
+				Optional:    true,
+				Computed:    true,
+				Description: "Instance database's name",
+			},
+			"type": {
+				Type:        schema.TypeString,
+				Required:    true,
+				ForceNew:    true,
```
[the remark this answers is not in the document]
Comment (jerome-quere) [reply scaleway/resource_rdb_instance_beta.go]: We may do it in another MR

**Comment by jerome-quere** [comment]
We need documentation page

File: scaleway/resource_rdb_instance_beta.go [hunk 2]
```diff
@@ -0,0 +1,206 @@
...
+				Required:    true,
+				Description: "Password for the first user",
+			},
+			"tags": {
+				Type: schema.TypeList,
+				Elem: &schema.Schema{
+					Type: schema.TypeString,
+				},
+				Optional:    true,
+				Description: "list of tags [\"tag1\", \"tag2\", ...] attached to a database instance",
```
Comment (loicbourgois) [inline scaleway/resource_rdb_instance_beta.go]: `List of tags ..`

With a capital letter.

File: scaleway/resource_rdb_instance_beta.go [hunk 3]
```diff
@@ -0,0 +1,206 @@
...
+				Required:    true,
+				ForceNew:    true,
+				Description: "Database's engine version id",
+			},
+			"is_ha_cluster": {
+				Type:        schema.TypeBool,
+				Optional:    true,
+				ForceNew:    true,
+				Default:     false,
+				Description: "Enable or disable high availability for database instance",
```
Comment (loicbourgois) [inline scaleway/resource_rdb_instance_beta.go]: for the database instance

File: scaleway/resource_rdb_instance_beta.go [hunk 4]
```diff
@@ -0,0 +1,206 @@
...
+			},
+			"disable_backup": {
+				Type:        schema.TypeBool,
+				Optional:    true,
+				Default:     false,
+				Description: "Disable automated backup on your database instance",
```
Comment (loicbourgois) [inline scaleway/resource_rdb_instance_beta.go]: for the database instance

File: scaleway/resource_rdb_instance_beta.go [hunk 5]
```diff
@@ -0,0 +1,206 @@
...
+			},
+			"user_name": {
+				Type:        schema.TypeString,
+				Required:    true,
+				Description: "Identifier of the first user of your database",
```
Comment (loicbourgois) [inline scaleway/resource_rdb_instance_beta.go]: Identifier for the first user of the database instance

File: scaleway/resource_rdb_instance_beta.go [hunk 6]
```diff
@@ -0,0 +1,206 @@
...
+			},
+			"password": {
+				Type:        schema.TypeString,
+				Required:    true,
+				Description: "Password for the first user",
```
Comment (loicbourgois) [inline scaleway/resource_rdb_instance_beta.go]: Password for the first user of the database instance

File: scaleway/resource_rdb_instance_beta.go [hunk 7]
```diff
@@ -0,0 +1,206 @@
...
+				Type:        schema.TypeString,
+				Optional:    true,
+				Computed:    true,
+				Description: "Instance database's name",
+			},
+			"type": {
+				Type:        schema.TypeString,
+				Required:    true,
+				ForceNew:    true,
+				Description: "The type of instance node you want to create",
```
Comment (loicbourgois) [inline scaleway/resource_rdb_instance_beta.go]: The type of database instance you want to create

File: scaleway/resource_rdb_instance_beta.go [hunk 8]
```diff
@@ -0,0 +1,206 @@
...
+		Importer: &schema.ResourceImporter{
+			State: schema.ImportStatePassthrough,
+		},
+		SchemaVersion: 0,
+		Schema: map[string]*schema.Schema{
+			"name": {
+				Type:        schema.TypeString,
+				Optional:    true,
+				Computed:    true,
+				Description: "Instance database's name",
```
Comment (loicbourgois) [inline scaleway/resource_rdb_instance_beta.go]: Name of the database instance

File: scaleway/resource_rdb_instance_beta.go [hunk 9]
```diff
@@ -0,0 +1,206 @@
...
+		Timeout:    InstanceServerWaitForTimeout,
+	})
+	if err != nil {
+		return err
+	}
+
+	return resourceScalewayRdbInstanceBetaRead(d, m)
+}
+
+func resourceScalewayRdbInstanceBetaRead(d *schema.ResourceData, m interface{}) error {
```
Comment (loicbourgois) [inline scaleway/resource_rdb_instance_beta.go]: what is `m`?

Comment (kindermoumoute) [reply scaleway/resource_rdb_instance_beta.go]: m is the common terraform variable for `meta` https://www.terraform.io/docs/extend/writing-custom-providers.html#defining-resources

**Pull request merged**